### DISCUSSION TITLE
Are resource constraints likely to be needed? add it to Deploy

### DISCUSSION THREAD

**Discussion by Euraxluo** [dora-rs/dora#335] [Q&A]
like this:
```rust
#[derive(Debug, Clone, Default, Serialize, Deserialize)]
#[serde(deny_unknown_fields)]
pub struct Deploy {
    /// machine 
    pub machine: Option<WorkId>,
    /// resource
    pub cpu: Option<f64>,
    pub gpu: Option<f64>,
    pub memory: Option<i64>, 
    /// worker num
    pub min_workers: Option<i64>,
    pub max_wowkers: Option<i64>,
}
```

**Comment by haixuanTao** [comment]
I think that it would be a great idea, but not sure of how to implement those limit. Do you have an implementation idea of how you would limit the consumption of CPU/Memory ?

**Comment by Euraxluo** [comment]
The idea came from Ray, but it also comes with a lot of extras, such as the need for more distributed coordination as the cluster gets resource control. I think we need to think from the point of view of the product, not technically difficult.

On UNIX, many tools can limit the resources of a process.
So there is also a cross-platform compatibility burden.

**Comment by meua** [comment]
We can consider learning container technology to limit resource usage by using cgroup and namespace. If you don’t consider getting on the bus, it’s better to borrow the existing mature technology, such as docker-compose or kubernetes.

**Reply by haixuanTao** [reply]
Please avoid using expression such as `If you don’t consider getting on the bus`. It tends to close the discussion. As mentioned there is many implementation possible. Free to consider any implementation.

请不用`If you don’t consider getting on the bus`的方法来沟通.  有很多方法可以做这个功能，都能接受的

**Reply by meua** [reply]
Ok, thanks for the reminder.
